crates/proto/src/rr/domain/label.rs

```diff
@@ -394,6 +401,70 @@ mod tests {
         assert_eq!(Label::from_utf8("🦀").unwrap().to_ascii(), "xn--zs9h");
     }
 
+    fn assert_panic_label_too_long(error: ProtoResult<Label>, len: usize) {
+        // poor man debug since ProtoResult don't implement Partial Eq due to ssl errors.
+        eprintln!("{:?}", error);
+        assert!(error.is_err());
+        match *error.unwrap_err().kind() {
+            ProtoErrorKind::LabelBytesTooLong(n) if n == len => (),
+            ProtoErrorKind::LabelBytesTooLong(e) => {
+                panic!(
+                    "LabelTooLongError error don't report expected size {} of the label provided.",
+                    e
+                )
+            }
+            _ => panic!("Should have returned a LabelTooLongError"),
+        }
+    }
+
+    #[test]
+    fn test_label_too_long_ascii_with_utf8() {
+        let label_too_long = "alwaystestingcodewithatoolonglabeltoolongtofitin63bytesisagoodhabit";
+        let error = Label::from_utf8(label_too_long);
+        assert_panic_label_too_long(error, label_too_long.len());
+    }
+
+    #[test]
+    fn test_label_too_long_utf8_puny_emoji() {
+        // too long only puny 65
+        let emoji_case = "💜🦀🏖️🖥️😨🚀✨🤖💚🦾🦿😱😨✉️👺📚💻🗓️🤡🦀😈🚀💀⚡🦄";
+        let error = Label::from_utf8(emoji_case);
+        assert_panic_label_too_long(error, 64);
+    }
+
+    #[test]
+    fn test_label_too_long_utf8_puny_emoji_mixed() {
+        // too long mixed 65
+        // Something international to say
+        // "Hello I like automn coffee 🦀 interresting"
+        let emoji_case = "こんにちは-I-mögen-jesień-café-🦀-intéressant";
+        let error = Label::from_utf8(emoji_case);
+        assert_panic_label_too_long(error, 65);
```

Comment:
Maybe it's better to compute puny length with an encoded string and avoid hardcoding it?

Reply:
It shouldn't change, so I think the static value is ok in a test.